### PR TITLE
Add recurse flag

### PR DESCRIPTION
**Pull request by KeenWill**

### CODE REVIEW

File: bin/camelot.ml [hunk 1]
```diff
@@ -51,34 +54,44 @@ let lex_src file =
...
-    else d ^ "/" in
   let dir = sanitize_dir dirname in
   if not (file_exists dir && is_directory dir) 
   then fail @@ dir ^ " doesn't exist or isn't a directory!";
   readdir dir |> Array.to_list |> List.map (fun file -> dir ^ file)
 
 let usage_msg =
+  "invoke with -r (only works if -d is set too) to recurse into subdirectories\n" ^
   "invoke with -d <dir_name> to specify a directory to lint, or just run the program with default args\n" ^
   "invoke with -show <student | ta | gradescope> to select the display type - usually ta's want a briefer summary" ^
```
Comment (Vighnesh-V) [inline bin/camelot.ml:71]:
```suggestion
  "invoke with -show <student | ta | gradescope> to select the display type - usually ta's want a briefer summary\n" ^
```

add a new line here for display purposes

File: bin/camelot.ml [hunk 2]
```diff
@@ -51,34 +54,44 @@ let lex_src file =
...
   "invoke with -f <.ml filename> to lint a particular file"
 
-let parse_sources_in dirname = 
+let rec parse_sources_in dirname : (string * Parsetree.structure) list =
```
Comment (Vighnesh-V) [inline bin/camelot.ml:74]: Separate the functionality of recursively finding for file names from actually lexing and parsing the files.

Perhaps make a ```let files_in_dir_rec ``` helper / merge with the files_in_dir function? Then use that inside of parse_sources_in, to determine the files that have to be parsed. This just helps separate the fxnality of parsing from the functionality of finding files :D

File: bin/camelot.ml [hunk 3]
```diff
@@ -29,7 +30,9 @@ let set_lint_file : string -> unit = fun s ->
 let spec =
   let open Arg in
   align [
-    "-d", Set_string lint_dir, 
+    "-r", Set recurse, 
+    "If calling on a directory using -d, recurse into its subdirectories"
```
Comment (Vighnesh-V) [inline bin/camelot.ml:34]:
```suggestion
    "\tIf calling on a directory using -d, recurse into its subdirectories"
```

File: bin/camelot.ml [hunk 4]
```diff
@@ -29,7 +30,9 @@ let set_lint_file : string -> unit = fun s ->
...
+  ;  "-d", Set_string lint_dir, 
     "Invoke the linter on the provided directory, defaulting to the current directory, non re"
```
Comment (Vighnesh-V) [inline bin/camelot.ml:36]:
```suggestion
    "\tInvoke the linter on the provided directory, defaulting to the current directory, non re"
```

File: bin/camelot.ml [hunk 5]
```diff
@@ -29,7 +30,9 @@ let set_lint_file : string -> unit = fun s ->
...
   ; "-show", String set_display_type,
     "Make the linter output display for either ta's | students | gradescope"
```
Comment (Vighnesh-V) [inline bin/camelot.ml:38]:
```suggestion
    "\tMake the linter output display for either ta's | students | gradescope"
```

File: bin/camelot.ml (hunk 2 above)
Comment (KeenWill) [reply bin/camelot.ml:74]: okay that sounds appropriate i'll implement this

Comment (KeenWill) [reply bin/camelot.ml]: done

**Pull request merged**